file_format.md

```diff
@@ -0,0 +1,197 @@
+  * version - 1.0
+  * msgType - always 1
+  * bundleStart - the frame index of the first frame in the bundle
+  * bundleSize - the number of frames in the bundle
+  * bundleData - a list of frames, for each frame:
+    * frameNumber - the index of this frame in the trajectory
+    * time - the simulated time at this frame
+    * data - a buffer of agent instance data, in order for each instance:
+      * visualization type
+        * 1000 = default, rendered with PDB or mesh
+        * 1001 = fiber, rendered as a line
+      * agent type ID - integer number ID for the agent’s type. This is used to look up its display data (name and geometry)
+      * position X, Y, Z - agent’s 3D position
+      * rotation X, Y, Z - euler angles for agent’s orientation
+      * radius - the radius the agent occupies, used for drawing a sphere
+      * subpoints - the number of proceeding values that are extra data belonging to this agent, followed by a list of numerical data
+        * for default type (1000), this is not used
+          * ex: subpoints = 0
+        * for fiber type (1001), this is the list of positions XYZ of the points along the fiber
+          * ex: subpoints = 9, pos1 X, pos1 Y, pos1 Z, pos2 X, pos2 Y, pos2 Z, pos3 X, pos3 Y, pos3 Z 
+* **plot data** - a list of plots, either scatterplots or histograms, for each:
+  * version - 1.0
+  * data - a list of plots, each has:
+    * layout
+      * title - label for the plot
+      * xaxis 
+        * title - the x-axis label
+        * if this label contains the text ‘time’, scatterplots will be considered time dependent and have a dynamic time bar rendered over them
+      * yaxis
+        * title - the y-axis label
+    * data - a list of traces, for each
+      * name - the label for this trace
+      * type - either 'scatter' or 'histogram'
+      * x - a list of x-values
+      * y - only for a scatterplot, a list of y-values, one for each x-value
+      * mode - only for a scatterplot, draw the data points as either 'lines' or 'markers', if not provided default to markers
+
+## Example Data
+```javascript
+{
+    // trajectory info
+    "trajectoryInfo" : {
+        "version" : 1,
+        // time
+        "timeStepSize" : 0.5,
+        "totalSteps" : 1000,
+        // box size
+        "size" : {
+            "x" : 300,
+            "y" : 300,
+            "z" : 300
+        },
+        // agent display data
+        "typeMapping": {
+            "0" : {
+                "name" : "agent1",
+                "pdb" : "agent1.pdb",  // optional
+                "mesh" : "agent1.obj"  // optional
+            },
+            "1" : {
+                "name" : "agent1#bound",
+                "pdb" : "agent1.pdb",  // optional
+                "mesh" : "agent1.obj"  // optional
+            },
+            "2" : {
+                "name" : "agent2",     // optional
```

Comment:
Is this optional?

Reply:
it does fallback to ID, but I didn't mean to add optional to this one